.pre-commit-config.yaml

```diff
@@ -0,0 +1,7 @@
+# forces black to format code before every commit d
+repos:
+  - repo: https://github.com/psf/black
+    rev: 23.3.0
+    hooks:
+      - id: black
```

Comment:
Also a question, does this format the code or just check the formatting?

I would prefer that this is a check that then forces the user to update on their side as otherwise your code changes when you push it. In very rare edge cases this could mean bugs are introduced due to "formatting" changes.

```yaml
      - uses: psf/black@stable
        with:
          options: "--check --diff --color"
```

for this to work you may need to do something like this prior to the `black` step:

```yaml
      - name: Ensure python3 venv is installed for Black
        run: |
          apt -y install python3-venv
```

Reply:
Good question. The hook only applies black on files that are in the staging area. My thought was when rubi-py is being worked on, black can automatically be added onto files that are being worked on. If the edge case bugs show up, the tests will fail. But I see how this is reliant on our testing coverage

Correct me if I'm wrong Adam. Add black as a github action -> if a file has not been formatted, fail the test and let the user know that the file must be formatted. 

Am good with either but I think the pre commit hook is a bit cleaner.